File: framework/data-jpa/src/test/java/top/bettercode/simpleframework/data/test/resp/AUser.java
package top.bettercode.simpleframework.data.test.resp;

import java.util.Objects;
import top.bettercode.lang.util.StringUtil;

/**
 * @author Peter Wu
 */
public class AUser {

  private Integer id;
  private String firstName;
  private LastName lastName;

  public AUser() {
  }

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getFirstName() {
    return firstName;
  }

  public void setFirstName(String firstName) {
    this.firstName = firstName;
  }


  public LastName getLastName() {
    return lastName;
  }

  public void setLastName(LastName lastName) {
    this.lastName = lastName;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof AUser)) {
      return false;
    }
    AUser cUsers = (AUser) o;
    return Objects.equals(firstName, cUsers.firstName) && Objects.equals(lastName,
        cUsers.lastName);
  }

  @Override
  public int hashCode() {
    return Objects.hash(firstName, lastName);
  }

  @Override
  public String toString() {
    return StringUtil.json(this);
  }
}

File: data-jpa/src/test/kotlin/top/bettercode/summer/data/jpa/resp/AUser.kt
package top.bettercode.summer.data.jpa.resp

import top.bettercode.summer.tools.lang.util.StringUtil.json
import java.util.*

/**
 * @author Peter Wu
 */
class AUser {
    var id: Int? = null
    var firstName: String? = null
    var lastName: LastName? = null
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is AUser) {
            return false
        }
        val cUsers = other
        return firstName == cUsers.firstName && lastName == cUsers.lastName
    }

    override fun hashCode(): Int {
        return Objects.hash(firstName, lastName)
    }

    override fun toString(): String {
        return json(this)
    }
}
